pos_access_right/static/src/js/models.js

```diff
@@ -0,0 +1,40 @@
+odoo.define("pos_access_right.models", function (require) {
+    "use strict";
+
+    var models = require("point_of_sale.models");
+    var posmodel_super = models.PosModel.prototype;
+
+    models.PosModel = models.PosModel.extend({
+        get_cashier: function () {
+            const pos_cashier = posmodel_super.get_cashier.apply(this);
+            const cashier = this.env.pos.users.find(
+                (user) => user.id === pos_cashier.user_id[0]
+            );
+            pos_cashier.hasGroupPayment =
+                cashier &&
+                cashier.groups_id.includes(this.env.pos.config.group_payment_id[0]);
+            pos_cashier.hasGroupDiscount =
+                cashier &&
+                cashier.groups_id.includes(this.env.pos.config.group_discount_id[0]);
+            pos_cashier.hasGroupNegativeQty =
+                cashier &&
+                cashier.groups_id.includes(
+                    this.env.pos.config.group_negative_qty_id[0]
+                );
+            pos_cashier.hasGroupPriceControl =
+                cashier &&
+                cashier.groups_id.includes(
+                    this.env.pos.config.group_change_unit_price_id[0]
+                );
+            pos_cashier.hasGroupMultiOrder =
+                cashier &&
+                cashier.groups_id.includes(this.env.pos.config.group_multi_order_id[0]);
+            pos_cashier.hasGroupDeleteOrder =
+                cashier &&
+                cashier.groups_id.includes(
+                    this.env.pos.config.group_delete_order_id[0]
+                );
+            return pos_cashier;
```

Comment:
I take it wasn't possible to extend the users.loaded function, to compute this on PoS startup for all users rather than doing it every time `get_cashier` is called?

Unfortunately doing it like this is worse than what you have before :(
Because `get_cashier` is called many many times across the point of sale, and you'd be adding an unnecessary overhead every time :(

I'm not blocking on this as the overhead isn't much, just nitpicking..

I'm curious to know why the proposed solution didn't work though :sweat_smile:

Reply:
At least we need to check cashier has proper access rights or not , in odoo we can set employee as cashier also so i have done like that.

Reply:
Yes, we need to check access right but the question is _where_.

My suggestion is to do it during PoS startup, right after the `res.users` are loaded, similarily to how [the core module computes the `role` property](https://github.com/odoo/odoo/blob/3780fa2af5d5f6cac91e419bcab69a253db280bd/addons/point_of_sale/static/src/js/models.js#L325-L343). See my comment there https://github.com/OCA/pos/pull/630#discussion_r637591752.

Alternatively, if my untested proposal doesn't work.. maybe `set_cashier` is a better place to compute this, as it's only called every once in a while